lib/API/JsonApi/v1/Resource/SuiteBeanResource.php

```diff
@@ -431,9 +440,147 @@ private function fromResource(Resource $resource)
      * @param Relationship $relationship
      * @return SuiteBeanResource
      */
-    public function withRelationship(\SuiteCRM\API\JsonApi\v1\Resource\Relationship $relationship) {
+    public function withRelationship(\SuiteCRM\API\JsonApi\v1\Resource\Relationship $relationship)
+    {
         $relationshipName = $relationship->getRelationshipName();
         $this->relationships[$relationshipName]['data'] = $relationship->toJsonApiResponse();
         return clone $this;
     }
+
+    /**
+     * Set the attributes to download a base64 encoded file
+     * @param \File $bean
+     * @param string $fieldName
+     * @throws \SuiteCRM\API\v8\Exception\ApiException
+     */
+    private function retrieveFileFromBean(\File $bean, $fieldName)
+    {
+        if (empty($bean->{$fieldName})) {
+            // return if a file has not been uploaded
+            return;
+        }
+
+        if ($bean instanceof \Document) {
+            $this->retrieveFileFromDocumentBean($bean, $fieldName);
+        } elseif ($bean instanceof \File) {
+            $this->retrieveFileFromFileBean($bean, $fieldName);
+        } else {
+            throw new ApiException(
+              '[SuiteBeanResource][retrieveFileFromBean][Unsupported Type] '.
+              $bean->module_name
+            );
+        }
+    }
+
+    /**
+     * @param \Note $bean
+     * @param string $fieldName
+     * @throws ApiException
+     */
+    private function retrieveFileFromFileBean(\Note $bean, $fieldName)
+    {
+        $fileFieldName = $fieldName . '_file';
+
+        $this->attributes[$fieldName] = $bean->{$fieldName};
+        // Get file
+        $file_path = \UploadStream::getDir() . '/' . $bean->id;
+
+        if (empty($file_path)) {
+            throw new ApiException(
+                '[SugarBeanResource] [retrieveFileFromFileBean][Unable to find file] Note: ' .
+                $bean->id
+            );
+        }
+
+        $file_contents = file_get_contents($file_path);
+
+        if ($file_contents === false) {
+            throw new ApiException(
+                '[SugarBeanResource] [retrieveFileFromFileBean][Unable to get file contents] FileName: ' .
+                $this->attributes[$fieldName]
+            );
+        }
+
+        $file = base64_encode($file_contents);
+        $this->attributes[$fileFieldName] = $file;
+    }
+
+    /**
+     * @param \Document $bean
+     * @param string $fieldName
+     * @throws ApiException
+     */
+    private function retrieveFileFromDocumentBean(\Document $bean, $fieldName)
```

Comment:
This method is almost the same like above. I believe you can merge them.

Reply:
Ok. I have merged into retrieveFileFromBean.